tests/clv/test_utils.py

```diff
@@ -137,7 +150,7 @@ def test_customer_lifetime_value_with_known_values(test_summary_data, fitted_bg)
         time=1,
         discount_rate=0.0,
     ).mean(("chain", "draw"))
-    np.testing.assert_almost_equal(clv_d0, expected, decimal=5)
+    np.testing.assert_almost_equal(clv_d0, expected, decimal=3)
```

Comment:
I think this test was completely deterministic (given the `rng`). In that case the values should be exactly the same as before your changes. This could mean somewhere we're not propagating some seed or something, which would be important to fix. 

The first thing to check is whether the test was indeed deterministic, meaning `clv_d0` and `clv_d1` was always the same.

Reply:
I've went through the model class changes and I didn't notice any seeds being passed in the class itself. also the customer_lifetime_value or subsequently called functions don't include one. If the test was completely deterministic, why use assert_almost_equal, instead assert_equal?

Reply:
because for float comparisons you never want to compare strict equality. Here the results are computer by us and lifetimes so they could differ slightly (even if deterministically)